### PR TITLE
Have generator create application.js file if not found

### LINKED ISSUE #78: Generator raises error when application.js is not found
We currently have our application split up into multiple smaller applications so don't use the main `application.js` file. It would nice if the generator allowed you to specify the name of the manifest file e.g.

```bash
rails generate react_on_rails:install -m my_cool_app
```

### PR DESCRIPTION
**Pull request by robwise**
Fixes #78
Generator used to put a setup error message when it did not find
application.js. Now, it will simply just create one with the
necessary data in it.

### CODE REVIEW

File: lib/generators/react_on_rails/base_generator.rb [hunk 1]
```diff
@@ -73,12 +73,12 @@ def update_application_js
...
-        application_js = dest_file_exists?(application_js_path) || dest_file_exists?(application_js_path + ".coffee")
-        if application_js
-          prepend_to_file(application_js, data)
+        app_js_path = "app/assets/javascripts/application.js"
+        found_app_js = dest_file_exists?(app_js_path) || dest_file_exists?(app_js_path + ".coffee")
+        if found_app_js
+          prepend_to_file(found_app_js, data)
         else
-          puts_setup_file_error("#{application_js} or #{application_js}.coffee", data)
+          create_file(app_js_path, data << "\n\n//= require turbolinks\n\n")
```
Comment (justin808) [inline lib/generators/react_on_rails/base_generator.rb]: Why add turbolinks? It's not a requirement of react_on_rails. We support it, but don't necessarily encourage it.

I'd say remove this.

Comment (robwise) [inline lib/generators/react_on_rails/base_generator.rb]: @justin808 roger that, makes sense

**Comment by robwise** [comment]
@justin808 this should be good to go pending a final review

**Comment by justin808** [comment]
@robwise This looks fine so long as it's well tested.

**Comment by robwise** [comment]
@justin808 yep, the important part of the associated test is here: https://github.com/shakacode/react_on_rails/pull/108/files#diff-dd34ce0d9a02be7e957b081e5d11cb7cR146 since it doesn't create an existing application.js like it usually does, and yet the file still shows up as expected as running the generator.

**Pull request merged**